sdk/python/lib/pulumi/runtime/settings.py

```diff
@@ -92,9 +86,44 @@ def __init__(
         else:
             self.engine = None
 
+    @contextproperty
+    def monitor(self) -> Optional[resource_pb2_grpc.ResourceMonitorStub]:
+        ...
+
+    @contextproperty
+    def engine(self) -> Optional[engine_pb2_grpc.EngineStub]:
+        ...
+
+    @contextproperty
+    def project(self) -> Optional[str]:
+        ...
+
+    @contextproperty
+    def stack(self) -> Optional[str]:
+        ...
+
+    @contextproperty
+    def parallel(self) -> Optional[bool]:
+        ...
+
+    @contextproperty
+    def dry_run(self) -> Optional[bool]:
+        ...
+
+    @contextproperty
+    def legacy_apply_enabled(self) -> Optional[bool]:
+        ...
+
+    @contextproperty
+    def feature_support(self) -> Optional[dict]:
+        ...
```

Comment:
Why do these individual attributes need to use contextvars? Instead, why not store the top-level `SETTINGS` var as a `ContextVar`, similar to what you're doing in #10472 for `CONFIG`, `ROOT`, and `_SECRET_KEYS`?

Reply:
The reason I didn't want to do that was throughout our codebase, the way and frequency in which we access SETTINGS is different than the others.  `CONFIG`, `ROOT`, and `_SECRET_KEYS` are simple objects which are accessed relatively infrequently.  In the case of settings, however, we don't use the getter functions universally, and access its attributes directly, so modifying the way in which our code interacts with SETTINGS throughout felt like a higher risk change than to maintain its interface and hide the complexity within the object itself.

Reply:
OK, fair enough. 😄